### PR TITLE
Add custom 500 error page

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Add custom error 500 page
- Refactor 404 error page and create reusable error page component

## Justify why they are needed

They are basically the same page.

Caveat: it's not 100% clear what should happen when user clicks "try again".

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1667** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1667" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1667?utm_source=stack-comment).

File: apps/store/public/locales/en/common.json [hunk 1]
```diff
@@ -1,13 +1,16 @@
 {
   "404_PAGE_BUTTON": "Go to start page",
   "404_PAGE_MESSAGE": "The page could not be found",
+  "500_PAGE_BUTTON": "Try again",
```
Comment (alebedev) [inline apps/store/public/locales/en/common.json:4]: I think this should be something like "Go to home page".  500 errors that we didn't catch are probably unrecoverable without at least full reload

**Review by alebedev — APPROVED** [review]
Looks good

**Comment by robinandeer** [comment]
@alebedev This is what would happen currently, the rest is just a copy question so let's merge and ask design

**Pull request merged**